lib/rules/no-unused-vars.js

```diff
@@ -126,6 +126,26 @@ module.exports = {
         }
 
         /**
+         * Determines if a variable has a sibling rest property
+         * @param {Variable} variable - EScope variable object.
+         * @returns {boolean} True if the variable is exported, false if not.
+         * @private
+         */
+        function hasRestSpreadSibling(variable) {
+            const matches = variable.defs.filter(def => def.name.type === "Identifier");
+            const lastProperty = matches[matches.length - 1];
+
+            return lastProperty && (
+                lastProperty.node.id &&
+                lastProperty.node.id.type === "ObjectPattern" &&
+                lastProperty.node.id.properties.some(property => (
+                    property.type === "ExperimentalRestProperty" ||
+                    property.type === "RestProperty"
+                ))
```

Comment:
I think you might have removed the wrong iteration here -- it's still necessary to keep the `filter` statement from before (if a variable has multiple declarations, the rule should check all of them), but the `node.id.properties.some()` iteration is unnecessary because a rest property will always be the last property of a given node.

I think it should look roughly like this:

```js
function hasRestSpreadSibling(variable) {
    return variable.defs
        .filter(def => def.name.type === "Identifier")
        .some(def => (
            def.node.id &&
            def.node.id.type === "ObjectPattern" &&
            def.node.id.properties.length &&
            new Set(["ExperimentalRestProperty", "RestProperty"]).has(def.node.id.properties[def.node.id.properties.length - 1].type)
        ));
}
```